src/lua/api-gateway/dogstatsd/Dogstatsd.lua

```diff
@@ -0,0 +1,100 @@
+
+--- Auxiliary module used to extract dogstats deamon initialization methods, incrementation of metrics
+-- and calls to other useful functions
+-- employed to measure calls to the Oauth provider: number, duration, etc.
+
+local Dogstatsd = {}
+
+function Dogstatsd:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+-- @return The loaded module, or nil if the module cannot be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        return require(module)
+    end
+
+    local res, cls = pcall(requiref, module)
+    if not (res) then
+        ngx.log(ngx.WARN, "Could not load module [", module, "].")
```

Comment:
You could log the error message. If res is false then cls is actually an error message.
 Another suggestion is to rename res to status and cls to result.